### PR TITLE
Add General CMake Build Option

### PR DESCRIPTION
**Pull request by ajpowelsnl**
Proposing adding CMake build option (`Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE`) to the list of general options

### CODE REVIEW

File: docs/source/keywords.rst [hunk 1]
```diff
@@ -160,6 +160,10 @@ General options
       * Aggressively vectorize loops
       * ``OFF``
 
+    * * ``Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE``
+      * Analogous to CUDA/HIP as CMake language
```
Comment (masterleinad) [inline docs/source/keywords.rst:164]: The doc string in `CMake` is
```suggestion
      * Whether to use native cmake language support
```

Comment (ajpowelsnl) [reply docs/source/keywords.rst:165]: Sounds good, but I deviated somewhat from your suggestion.  The "use native CMake ..." is the more general, so I put it ahead of the other statement.

**Comment by masterleinad** [comment]
https://github.com/kokkos/kokkos-core-wiki/pull/555/files is the more comprehensive pull request. Can you instead comment there?

**Comment by masterleinad** [comment]
Superseded by #555.

**Pull request closed without merging**